homeassistant/components/vacuum/xiaomi_miio.py

```diff
@@ -82,6 +84,20 @@
                  SUPPORT_SEND_COMMAND | SUPPORT_LOCATE | \
                  SUPPORT_STATUS | SUPPORT_BATTERY | SUPPORT_CLEAN_SPOT
 
+STATE_CODE_TO_STATE = {
+    5: STATE_CLEANING,
+    11: STATE_CLEANING,
+    16: STATE_CLEANING,
+    17: STATE_CLEANING,
+    8: STATE_DOCKED,
+    10: STATE_PAUSED,
+    3: STATE_IDLE,
+    6: STATE_RETURNING,
+    15: STATE_RETURNING,
+    9: STATE_ERROR,
+    12: STATE_ERROR,
+}
```

Comment:
Would it make sense to order this by the key, so that one would see what items are missing? e.g. 4 is "remote control active" or 7 is "manual mode", which are missing here, see https://github.com/rytilahti/python-miio/blob/master/miio/vacuumcontainers.py#L53 .

Furthermore it may be useful to (at least debug) log the missing state code, maybe even print out the state and mention that it is not handled?